neo4j-gremlin/src/main/java/org/apache/tinkerpop/gremlin/neo4j/structure/Neo4jHelper.java

```diff
@@ -45,13 +44,16 @@ public static boolean isDeleted(final Neo4jNode node) {
         try {
             node.getKeys();
             return false;
-        } catch (final IllegalStateException e) {
-            return true;
+        } catch (final RuntimeException e) {
+            if (isNotFound(e))
+                return true;
+            else
+                throw e;
         }
     }
 
-    public static boolean isNotFound(RuntimeException ex) {
-        return ex.getClass().getSimpleName().equals("NotFoundException");
+    public static boolean isNotFound(final RuntimeException ex) {
+        return ex.getClass().getSimpleName().equals(NOT_FOUND_EXCEPTION);
```

Comment:
There's probably a good reason for the `toString()` comparison, but wouldn't it be easier/cleaner to do this:

```java
try {
    node.getKeys();
    return false;
} catch (NotFoundException e) {
    return true;
}
```